metrics/timer.js

```diff
@@ -2,6 +2,17 @@ var Meter = require('./meter'),
     Histogram = require('./histogram'),
     ExponentiallyDecayingSample = require('../stats/exponentially_decaying_sample');
 
+var MILLIS_IN_SEC = 1e3;
+var MILLIS_IN_NANO = 1e-6;
+var TimerContext = function(timer) {
+  this.timer = timer;
+  this.start = process.hrtime();
+}
+TimerContext.prototype.stop = function() {
+  var duration = process.hrtime(this.start);
+  this.timer.update(duration[0] * MILLIS_IN_SEC + duration[1] * MILLIS_IN_NANO);
```

Comment:
One issue with this is that its currently implied by the reporting interface that `Timer` is recording values in milliseconds.  In general, I think this is what people will want, but we lose precision here if anyone wants something more high resolution.

I think for now this is fine, i'll add some documentation that recordings are measured in milliseconds.

In a future major release we should change this to report in nanoseconds and also allow reporters to configure their duration unit (as detailed in #39), but for now I think milliseconds is fine.